### PR TITLE
devtools-archlinuxcn: update to 1.0.0

### PR DESCRIPTION
**Pull request by yan12125**
PKGBUILD is synced again from
https://gitlab.archlinux.org/archlinux/packaging/packages/devtools/

Rebase conflicts:

* Change default BUILDTOOL to devtools-archlinuxcn
* Use x86_64 mirrors for x86_64_v3

Those two commits are tested.

---

There are many [upstream commits](https://github.com/archlinux/devtools/compare/20230307...1.0.0). Most of them are related to git migration. There is one known issue: `offload-build` may fail if the server and the client has different devtools versions [1]

[1] https://gitlab.archlinux.org/archlinux/devtools/-/issues/108

### CODE REVIEW

**Review by lilydjwg — APPROVED** [review]
It's quite a lot changes. I hope things will go well.

**Comment by yuyichao** [comment]
Seems that there’s a change in the repo names? I assume some configure files needs to be updated?

**Comment by lilydjwg** [comment]
.conf files moved to `/usr/share/devtools/pacman.conf.d`. devtools-cn-git and devtools-arm-git needs to be updated too.

**Comment by yan12125** [comment]
Good point! Also, devtools-riscv64 needs a fix as well:
```
patching file makepkg-riscv64.conf (read from /usr/share/devtools/makepkg-x86_64.conf)
Hunk #1 FAILED at 35.
1 out of 1 hunk FAILED -- saving rejects to file makepkg-riscv64.conf.rej
```
/cc @felixonmars

**Comment by CoelacanthusHex** [comment]
> Good point! Also, devtools-riscv64 needs a fix as well:
> 
> ```
> patching file makepkg-riscv64.conf (read from /usr/share/devtools/makepkg-x86_64.conf)
> Hunk #1 FAILED at 35.
> 1 out of 1 hunk FAILED -- saving rejects to file makepkg-riscv64.conf.rej
> ```
> 
> /cc @felixonmars

I think devtools-riscv64 should be updated after Felix Yan does extra-community merge on RISC-V side manually.
After that, @Xeonacid will update AUR and lilac will pull automatically.

**Comment by yan12125** [comment]
I started a branch https://github.com/yan12125/devtools-cn/tree/devtools-updates, which fixes x86-64 chroot build scripts. However, ARM and i686 scripts are not touched. For Arch Linux 32, merging of [extra] and [community] is a work in progress [1], and I didn't find anything about that for Arch Linux ARM. Anyone knows what's going on for Arch Linux ARM?

[1] https://bbs.archlinux32.org/viewtopic.php?pid=9767

**Comment by yan12125** [comment]
Can someone test this patch for devtools-arm-git? I have no access to Arch Linux ARM devices.
```Diff
diff --git a/archlinuxcn/devtools-arm-git/PKGBUILD b/archlinuxcn/devtools-arm-git/PKGBUILD
index eb0738a..0f5b000 100644
--- a/archlinuxcn/devtools-arm-git/PKGBUILD
+++ b/archlinuxcn/devtools-arm-git/PKGBUILD
@@ -9,8 +9,10 @@ depends=('devtools')
 makedepends=('cmake' 'git')
 provides=('devtools-arm')
 conflicts=('devtools-arm')
-source=('git+https://github.com/yuyichao/devtools-arm')
-sha256sums=('SKIP')
+source=('git+https://github.com/yuyichao/devtools-arm'
+        'devtools-updates.diff')
+sha256sums=('SKIP'
+            '675f8571e17f8179c5a6a52d49e3859896e787de31028ee240fa18dfa96727ac')
 # People doing cross compilation with qemu and alike might want to override these
 backup=(usr/share/devtools/setarch-aliases.d/armv6h
         usr/share/devtools/setarch-aliases.d/armv7h)
@@ -21,6 +23,12 @@ pkgver() {
   echo "0.0.0.$(git rev-list --count HEAD).$(git rev-parse --short HEAD)"
 }
 
+prepare() {
+  cd devtools-arm
+
+  patch -Np1 -i ../devtools-updates.diff
+}
+
 build() {
   cd devtools-arm
   mkdir -p build
diff --git a/archlinuxcn/devtools-arm-git/devtools-updates.diff b/archlinuxcn/devtools-arm-git/devtools-updates.diff
new file mode 100644
index 0000000..b9d56c0
--- /dev/null
+++ b/archlinuxcn/devtools-arm-git/devtools-updates.diff
@@ -0,0 +1,24 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index fa7497e..e6a9aa3 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -21,12 +21,14 @@ endfunction()
+ 
+ function(install_arch CARCH CHOST CARCHFLAGS)
+   install_build_script("extra-${CARCH}-build")
+-  configure_file(pacman-extra.conf.in "pacman-extra-${CARCH}.conf" @ONLY)
+-  configure_file(makepkg.conf.in "makepkg-${CARCH}.conf" @ONLY)
++  configure_file(pacman-extra.conf.in "extra-${CARCH}.conf" @ONLY)
++  configure_file(makepkg.conf.in "${CARCH}.conf" @ONLY)
+   install(FILES
+-    "${CMAKE_CURRENT_BINARY_DIR}/pacman-extra-${CARCH}.conf"
+-    "${CMAKE_CURRENT_BINARY_DIR}/makepkg-${CARCH}.conf"
+-    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/devtools/")
++    "${CMAKE_CURRENT_BINARY_DIR}/extra-${CARCH}.conf"
++    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/devtools/pacman.conf.d/")
++  install(FILES
++    "${CMAKE_CURRENT_BINARY_DIR}/${CARCH}.conf"
++    DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/devtools/makepkg.conf.d/")
+ endfunction()
+ 
+ install_arch(arm armv5tel-unknown-linux-gnueabi "-march=armv5te ")
```

**Comment by yan12125** [comment]
Thanks! I created https://github.com/yuyichao/devtools-arm/pull/2 for devtools-arm changes.

Meanwhile, I updated this branch to devtools 1.0.1. There are no rebase conflicts and [upstream commits](https://github.com/archlinux/devtools/compare/1.0.0...v1.0.1) seem minimal.

**Comment by yuyichao** [comment]
All three PRs looks good to me. Is there any other change that's needed? I could merge all three at the same time or wait for someone to merge this before merging the two on my repo.

**Review by yuyichao — APPROVED** [review]

**Comment by yan12125** [comment]
> Is there any other change that's needed?

I believe no. devtools-riscv64 is already updated (https://github.com/archlinuxcn/repo/commit/d48f8e1a2e96f9b55bc3cad96226d8c57dfd4de4).

> I could merge all three at the same time

This sounds better.

**Comment by yuyichao** [comment]
OK. I’ll merge in a bit (after lunch…) feel free to remind me,

**Pull request merged**
